packages/pwa-kit-react-sdk/bin/upload.js

```diff
@@ -62,10 +62,19 @@ try {
     Utils.fail(message)
 }
 
+// Set the deployment target env var, this is required to ensure we
+// get the correct configuration oject.
+process.env.DEPLOY_TARGET = argv.t
+
+// Get the config object for the current target, if no target was set
+// it will return the defualt configuration and that one will be
+// validated.
+const config = Utils.getConfig()
```

Comment:
instead of passing the target implicitly thru environment variable, would it be better if we add an optional argument to `getConfig`, something like:
```suggestion

// Get the config object for the current target, if no target was set
// it will return the defualt configuration and that one will be
// validated.
const config = Utils.getConfig(argv.t)
```

Reply:
I kind of touched on this above a little. I'm a little on the fence. If we do something like passing in a argument, it means that the description of the function argument is somwthing like 'moduleName: is the file name of the configuration you want to load, this argument isn't used on the client and will be ignored'. 

Another argument for keeping things the way they are, is that if we pass in an argument for the file to load, it somewhat devalues to structure of the 'searchPlaces` option, a user can then use this method to load any config file they want. When we are really setting up rules to configuration saying, I'll load the config for your environment, if thats not found I'll load x, y, then z. 

If we are to change the function to accept a file name, it is more like this function loads a json file with the name you provide from the config folder, if it doesn't exist we'll load x, y, then z. 

I like dependency injection, but I think in this case we want the simple "loads the config for your environment" description.